packages/sdk-metrics/src/aggregator/types.ts

```diff
@@ -109,14 +109,14 @@ export interface Aggregator<T> {
   /**
    * Returns the {@link MetricData} that this {@link Aggregator} will produce.
    *
-   * @param descriptor the metric instrument descriptor.
+   * @param descriptor the metric descriptor.
    * @param aggregationTemporality the temporality of the resulting {@link MetricData}
    * @param accumulationByAttributes the array of attributes and accumulation pairs.
    * @param endTime the end time of the metric data.
    * @return the {@link MetricData} that this {@link Aggregator} will produce.
    */
   toMetricData(
-    descriptor: InstrumentDescriptor,
+    descriptor: MetricDescriptor,
```

Comment:
It seems like we don't need to repeat `toExternal` in every aggregator.

```suggestion
    descriptor: Descriptor,
```

Reply:
My initial goal with putting it into the individual `Aggregator`s was to ensure that we use the deprecated type as little as possible in the existing codebase. Here, we'd have to change it to `InstrumentDescriptor` which will cause a deprecation warning in each Aggregator, as `Descriptor` does not expose the necessary information anymore. :/

When we go to 2.0, we'd be able to pass `Descriptor` here, and introduce another argument `allowsNegativeValues: boolean` so that we can properly handle histogram sums (which need to be dropped, if false).